model/content_entity.go

```diff
@@ -749,3 +751,44 @@ func (w *ContentEntity) UnmarshalJSON(data []byte) error {
 	err = json.Unmarshal(data, &w.Property)
 	return err
 }
+
+//GenerateID adds a generated id to the payload based on the schema
+func GenerateID(payload []byte, entityFactory EntityFactory) ([]byte, error) {
```

Comment:
This should be a function ON a content entity that is maybe triggered on an isValid call. The isvalid could check to see if the identifier fields have value and if the identifier is one field then trigger the generateID function